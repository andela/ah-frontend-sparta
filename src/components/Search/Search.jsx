import React from 'react';

const Search = ({ handleChange, query, handleSearch }) => (
  <>
    <div className="search-container">
      <form onSubmit={handleSearch}>
        <input
          type="text"
          placeholder="Search by Article title"
          name="query"
          id="searchInput"
          defaultValue={query}
          onChange={handleChange}
        />
        <button type="submit"><i className="fa fa-search" /></button>
      </form>
    </div>
  </>
);


export default Search;
