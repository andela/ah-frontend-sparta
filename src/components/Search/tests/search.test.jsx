import React from 'react';
import renderer from 'react-test-renderer';
import Search from '../Search';

describe('Search page', () => {
  it('should render without failing', () => {
    const wrapper = renderer.create(<Search />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
