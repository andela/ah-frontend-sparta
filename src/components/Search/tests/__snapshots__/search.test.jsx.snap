// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search page should render without failing 1`] = `
<div
  className="search-container"
>
  <form>
    <input
      id="searchInput"
      name="query"
      placeholder="Search by Article title"
      type="text"
    />
    <button
      type="submit"
    >
      <i
        className="fa fa-search"
      />
    </button>
  </form>
</div>
`;
